userspace/libsinsp/k8s_event_handler.cpp

```diff
@@ -114,13 +116,35 @@ bool k8s_event_handler::handle_component(const Json::Value& json, const msg_data
 					if(!involved_object.isNull())
 					{
 						bool is_aggregate = (get_json_string(json , "message").find("events with common reason combined") != std::string::npos);
-						time_t last_ts = get_epoch_utc_seconds(get_json_string(json , "lastTimestamp"));
+						time_t last_ts = 0;
 						time_t now_ts = get_epoch_utc_seconds_now();
+						// So we first are going to check for "eventTime"
+						// If that is empty, we will check for "lastTimestamp"
+						// If that is also empty, use current timestamp and log it. 
+						// This change is necessiated because in v1beta1/events, "EventTime" is
+						// the main field that holds timestamp and `lastTimestamp` is deprecated.
+						// This change is addressed towards that. 
+						std::string evtTime = get_json_string(json, "eventTime");
+						std::string ts = get_json_string(json , "lastTimestamp");
+						if(!evtTime.empty())
+						{
+							last_ts	= get_epoch_utc_seconds(evtTime);
+						}
+						else if(!ts.empty())
+						{
+							last_ts = get_epoch_utc_seconds(ts);
+						}
+						else
+						{
+							// Ideally we should NEVER hit this case. But log it if we do, so we know.
+							g_logger.log("K8s EVENT: both eventTime and lastTimestamp are null, using current timestamp", sinsp_logger::SEV_INFO);
```

Comment:
we should add some detail about the event so we can diagnose how this can happen and/or attempt a repro. Just dumping the json here is fine as well imo.

Reply:
Added a json dump of the event.